longest-substring-without-repeating-character.js

```diff
@@ -0,0 +1,15 @@
+var lengthOfLongestSubstring = function (s) {
+  if (!s) return 0;
+  let max = 1;
+  for (let i = 0; i < s.length; i++) {
+    //optimized a bit, we don't need to go further if our maximum is already more than number of characters left
+    if (s.length - i < max) break;
+    for (let j = i + 1; j <= s.length; j++) {
+      const curMax = new Set(s.slice(i, j).split(""));
```

Comment:
Is `slice` increasing a runtime, to O(n^3) ?

Reply:
yes, I agree, slice is constant, but making set is n, so, n3 in total.